test/OpenTelemetry.Tests/Logs/LoggerProviderTests.cs

```diff
@@ -0,0 +1,117 @@
+// limitations under the License.
+// </copyright>
+#if !NET461
+using System;
+using System.Collections.Generic;
+
+using Microsoft.Extensions.Logging;
+
+using OpenTelemetry.Exporter;
+
+using OpenTelemetry.Resources;
+
+using Xunit;
+
+namespace OpenTelemetry.Logs.Tests
+{
+    public sealed class LoggerProviderTests
+    {
+        [Fact]
+        public void VerifyDefaultBehavior()
+        {
+            InitializeLoggerFactory(out OpenTelemetryLoggerProvider provider);
+
+            Assert.Contains(provider.GetResource().Attributes, (kvp) => kvp.Key == "service.name" && kvp.Value.ToString() == "unknown_service:testhost");
+        }
+
+        [Fact]
+        public void VerifyResourceBuilderAddService()
+        {
+            InitializeLoggerFactory(out OpenTelemetryLoggerProvider provider, configure: options => options.SetResourceBuilder(ResourceBuilder.CreateDefault().AddService(serviceName: "MyService", serviceVersion: "1.2.3")));
+
+            Assert.Contains(provider.GetResource().Attributes, (kvp) => kvp.Key == "service.name" && kvp.Value.ToString() == "MyService");
+            Assert.Contains(provider.GetResource().Attributes, (kvp) => kvp.Key == "service.version" && kvp.Value.ToString() == "1.2.3");
+        }
+
+        [Fact]
+        public void VerifyResourceBuilder_WithServiceNameEnVar()
+        {
+            try
+            {
+                Environment.SetEnvironmentVariable(OtelServiceNameEnvVarDetector.EnvVarKey, "MyService");
+
+                InitializeLoggerFactory(out OpenTelemetryLoggerProvider provider);
+
+                Assert.Contains(provider.GetResource().Attributes, (kvp) => kvp.Key == "service.name" && kvp.Value.ToString() == "MyService");
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable(OtelServiceNameEnvVarDetector.EnvVarKey, null);
+            }
+        }
+
+        [Fact]
+        public void VerifyResourceBuilder_WithAttributesEnVar()
+        {
+            try
+            {
+                Environment.SetEnvironmentVariable(OtelEnvResourceDetector.EnvVarKey, "Key1=Val1,Key2=Val2");
+
+                InitializeLoggerFactory(out OpenTelemetryLoggerProvider provider);
+
+                Assert.Contains(provider.GetResource().Attributes, (kvp) => kvp.Key == "Key1" && kvp.Value.ToString() == "Val1");
+                Assert.Contains(provider.GetResource().Attributes, (kvp) => kvp.Key == "Key2" && kvp.Value.ToString() == "Val2");
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable(OtelEnvResourceDetector.EnvVarKey, null);
+            }
+        }
+
+        private static void InitializeLoggerFactory(out OpenTelemetryLoggerProvider provider, Action<OpenTelemetryLoggerOptions> configure = null)
+        {
+            var exporter = new InMemoryExporter<LogRecord>(new List<LogRecord>());
+            var loggerFactory = LoggerFactory.Create(builder =>
```

Comment:
Where is this disposed?

Reply:
previous version of this method was returning the Factory.
It's not needed anymore and I've added a `using` to dispose in this method.